### PR TITLE
Using MP cookie to persist Variants Served

### PR DESCRIPTION
**Pull request by Erfaricy**
Just updated the readme from yesterdays thread.

### CODE REVIEW

**Comment by bentona** [comment]
Please remove .DS_Store file

**Comment by bentona** [comment]
Missing a lot of semi-colons, not sure how @jakiestfu feels about that.

**Comment by jakiestfu** [comment]
GET OUT OF HERE @bentona I'M STILL HURT

**Comment by jakiestfu** [comment]
> Please remove .DS_Store file
> Missing a lot of semi-colons

@Erfaricy Have you hinted the script?

**Comment by Erfaricy** [comment]
Lol damn ds_store.

No I haven't hinted the script, but if you want to point one out to me for Vim. Im open.

File: src/scenario.js [hunk 1]
```diff
@@ -75,10 +84,22 @@
...
+
+            if (!utils.checkVariant(scenarioOpts.name)) {
+                var chosenTestIndex = utils.chooseWeightedItem();
+                var test = self.tests[scenarioOpts.name][chosenTestIndex];
+                utils.setVariant(test.name);
+            } else {
+                var test = utils.checkVariant(scenarioOpts.name);
+                /* This is to find the callback for the test */
+                var tests = self.tests[scenarioOpts.name];
+                for (var i = 0; i <  tests.length; i++) {
```
Comment (bentona) [inline src/scenario.js]: `for(i in values)` is preferred when doing basic array iterations.

**Comment by dasnixon** [comment]
@jakiestfu any chance this PR is going to get merged? Anything that needs to be fixed?

**Comment by jakiestfu** [comment]
@mpalmerlee Will you please merge this pull request?

**Comment by dasnixon** [comment]
Actually, playing with this I'm not sure if this is functioning correctly

`var test = utils.checkVariant(scenarioOpts.name);` returns a string

Then tries to do `d.body.className += " "+test.className;`

I just forked it and will see if I discover anything, maybe I just am not sure how this should work :confused:

**Pull request merged**